actix-multipart/src/server.rs

```diff
@@ -768,7 +767,7 @@ impl PayloadBuffer {
             .map(|idx| self.buf.split_to(idx + line.len()).freeze());
 
         if res.is_none() && self.eof {
-            Err(MultipartError::Incomplete)
+            Ok(self.buf.take().freeze())
```

Comment:
Mismatched types error here, probably `Ok(Some(self.buf.take().freeze()))`.

Reply:
i dont think this is right code. this could lead to partial reads reported as valid

Reply:
Yep good call, ive moved this out into a new function `readline_or_eof` which is called by `read_boundary`, a partial read should then result in a `BoundaryError`